mod/execution/pkg/client/engine.go

```diff
@@ -61,7 +61,8 @@ func (s *EngineClient[
 			s.metrics.incrementNewPayloadTimeout()
 		}
 		return nil, s.handleRPCError(err)
-	} else if result == nil {
```

Comment:
not strictly necessary for this PR, but I like reducing indentation as much as possible